api/nomics.py

```diff
@@ -0,0 +1,125 @@
+        "website": "https://bitshares.org/",
+        "twitter": "example",
+        "capability": {
+            "markets": True,
+            "trades": True,
+            "tradesSocket": False,
+            "orders": False,
+            "ordersSocket": False,
+            "ordersSnapshot": True,
+            "candles": False
+        }
+    }
+
+
+def markets():
+    result = []
+    top_markets = bitshares_ws_client.request('database', 'get_top_markets', [100])
+
+    for market in top_markets:
+        result.append({
+            'id': market["base"] + "_" + market["quote"],
+            'base': market["base"],
+            'quote': market["quote"]
+        })
+    return result
+
+
+@cache.memoize()
+def trades(market, since):
+
+    market_id = market.split('_')
+    base = market_id[0]
+    quote = market_id[1]
+
+    now = datetime.datetime.now()
+    start = now
+    stop = now - datetime.timedelta(days=3)
+
+    results = []
+
+    _trades = bitshares_ws_client.request('database', 'get_trade_history',
+                                          [base, quote, start.strftime("%Y-%m-%dT%H:%M:%S"),
+                                           stop.strftime("%Y-%m-%dT%H:%M:%S"), 100])
+
+    for trade in _trades:
+        results.append({
+            "id": trade["sequence"],
+            "timestamp": trade["date"],
+            "price": trade["price"],
+            "amount": trade["amount"],
+            "order": "",
+            "type": "",
+            "side": "",
+            "raw": ""
+        })
+
+    while len(_trades) == 100:
+        start_seq = _trades[99]["sequence"]
+        _trades = bitshares_ws_client.request('database', 'get_trade_history_by_sequence',
+                                              [base, quote, start_seq, stop.strftime("%Y-%m-%dT%H:%M:%S"), 100])
+        for trade in _trades:
+            results.append({
+                "id": trade["sequence"],
+                "timestamp": trade["date"],
+                "price": trade["price"],
+                "amount": trade["amount"],
+                "order": "",
+                "type": "",
+                "side": "",
+                "raw": ""
+            })
+
+    if not since:
+        return list(reversed(results))[0:100]
+    else:
+        new_results = []
+        for r in results:
+            if int(r["id"]) > int(since):
+                new_results.append(r)
+        return list(reversed(new_results))[0:100]
+
+
+def snapshot(market):
+    result = {}
+
+    market_id = market.split('_')
+    base = market_id[0]
+    quote = market_id[1]
+
+    order_book = api.explorer.get_order_book(base, quote, 100)
+
+    bids = []
+    asks = []
+
+    for bid in order_book["bids"]:
+        bids.append([bid["price"], bid["base"]])
+
+    result["bids"] = bids
+
+    for ask in order_book["asks"]:
+        asks.append([ask["price"], ask["base"]])
+
+    result["asks"] = asks
+
+    result["timestamp"] = api.explorer.get_last_block_time()
```

Comment:
This is dangerous.
As we do 2 network queries the returned time could be a wrong order book time.
It's a flaw in the core API imho. We should fill a core issue for the long term.

Reply:
return an `timestamp` field in `get_order_book` node call among with the `asks` and `bids` ?

Reply:
yes exactly

Reply:
Returning utc time not with a second call but just with python from the server the api is running on. Adding timestamp to the node do not makes too much sense so discarding it by now.